src/FileStorage/DataTransformer.php

```diff
@@ -73,7 +73,7 @@ public function entityToFileObject(EntityInterface $entity): FileInterface
     public function fileObjectToEntity(FileInterface $file, ?EntityInterface $entity): EntityInterface
     {
         $data = [
-            'id' => $file->uuid(),
+            'id' => $file->uuid(), //FIXME
```

Comment:
Do we need to set the id for primary key records?

Reply:
The File object doesn't care, because the whole lib doesn't care about persistance (its not their job) but if you want to persist it and your file path relies on it, then you clearly should set it. The File objects only purpose and reason that it features certain things is that it might need them for generation pathes or doing certain operations in processors.

The File object is like a DTO between persistence that the user has to take core of in his app and the library that only knows about what the file object provides.